lib/streamlit/runtime/app_session.py

```diff
@@ -167,6 +167,11 @@ def __init__(
 
         self._fragment_storage: FragmentStorage = MemoryFragmentStorage()
 
+        # Counter for the number of ForwardMsgs sent by this AppSession. We don't
+        # currently do anything with this information, but we may eventually use it to
+        # allow us to replay ForwardMsgs dropped during a websocket disconnect.
+        self._forward_msg_counter = 0
```

Comment:
Maybe I am overlooking it, but I guess we can reset the counter after each full app rerun, can't we?

Reply:
I think it's generally more common for these types of sequence counters to not be reset since there may be some edge cases where doing so causes unexpected behavior, and there's little to no added cost to just having the counter be monotonically increasing (at a few million script runs, which scripts will rarely if ever get to, we'll just be sending a few extra bytes).

Reply:
My thinking was that keeping it incrementing _may_ lead people who are looking at it in the future to think that there is some cross-app run meaning to it. I feel resetting it to `0` would make it more obvious that there is nothing going on between full-app runs and that there are no side-effects.
But if we assume there could be unexpected behavior if we reset it (so there actually already might be side-effects), then I agree that we should just let it increment.

Reply:
My thinking was that keeping it incrementing may lead people who are looking at it in the future to think that there is some cross-app run meaning to it. I feel resetting it to 0 would make it more obvious that there is nothing going on between full-app runs and that there are no side-effects.
But if we assume there could be unexpected behavior if we reset it (so there actually already might be side-effects), then I agree that we should just let it increment.